.github/workflows/prover-test.yaml

```diff
@@ -0,0 +1,48 @@
+name: "Prover Test"
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - devnet
+      - testnet
+      - auto
+      - canary
+
+env:
+  HAS_BUILDPULSE_SECRETS: ${{ secrets.BUILDPULSE_ACCESS_KEY_ID != '' && secrets.BUILDPULSE_SECRET_ACCESS_KEY != '' }}
+  HAS_DATADOG_SECRETS: ${{ secrets.DD_API_KEY != '' }}
+  CARGO_INCREMENTAL: "0"
+  CARGO_TERM_COLOR: always
+
+# cancel redundant builds
+concurrency:
+  # cancel redundant builds on PRs (only on PR, not on branches)
+  group: ${{ github.workflow }}-${{ (github.event_name == 'pull_request' && github.ref) || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  prover-test:
+    runs-on: high-perf-docker
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # get all the history because cargo xtest --change-since origin/main requires it.
+      - uses: ./.github/actions/rust-setup
+      - name: install Prover dependencies
+        shell: bash
+        run: scripts/dev_setup.sh -b -p -y
+      - name: prepare move lang prover tooling.
+        shell: bash
+        run: |
+          # By setting these values the dev-setup.sh script can detect already installed executables (and check versions).
```

Comment:
Based on the comment...shouldn't these command be executed before `dev_setup.sh`?

Reply:
I think this is ok to go after `dev_setup.sh` because we are installing these dependencies on a fresh (CI testing) machine where Boogie and Z3 are not installed already. I removed the comment line because it is confusing and unhelpful here.